Sources/SwiftSyntaxBuilderGeneration/Templates/BuildableCollectionNodesFile.swift

```diff
@@ -0,0 +1,283 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SwiftSyntax
+import SwiftSyntaxBuilder
+
+let buildableCollectionNodesFile = SourceFile {
+  ImportDecl(
+    leadingTrivia: .docLineComment(copyrightHeader),
+    path: "SwiftSyntax"
+  )
+
+  for node in SYNTAX_NODES where node.isSyntaxCollection {
+    let type = node.type
+    let elementType = node.collectionElementType
+    let conformances = ["ExpressibleByArrayLiteral", "SyntaxBuildable", type.expressibleAs]
+    
+    // Generate collection node struct
+    StructDecl(
+      leadingTrivia: node.documentation.isEmpty
+        ? []
+        : .docLineComment("/// \(node.documentation)") + .newline,
+      modifiers: [TokenSyntax.public],
+      identifier: type.buildable,
+      inheritanceClause: createTypeInheritanceClause(conformances: conformances)
+    ) {
+      VariableDecl(.let, name: "elements", type: ArrayType(elementType: elementType.buildable))
+
+      // Generate initializers
+      createArrayInitializer(type: type, elementType: elementType)
+      createCombiningInitializer(type: type)
+      createArrayLiteralInitializer(elementType: elementType)
+
+      // Generate function declarations
+      createBuildFunction(node: node, type: type, elementType: elementType)
+      createBuildSyntaxFunction(type: type)
+      createExpressibleAsCreateFunction(type: type)
+      createSyntaxBuildableCreateFunction(type: type)
```

Comment:
A question regarding style: I have passed `type`, `elementType` and `node` separately to these functions for now. Since `type` and `elementType` can easily be derived from just `node`, would it be better to just pass `node`?

Reply:
I don’t think there’s a reason we need to pass `type` and `elementType` if they can be derived from `node` and there’s no reason you ever wanted to pass e.g. a different `type` than `node.type`.